dig.go

```diff
@@ -498,6 +523,18 @@ func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
 	return nil
 }
 
+func (c *Container) verifyAcyclic() error {
+	visited := make(map[key]struct{})
+	for _, n := range c.nodes {
+		if err := detectCycles(n, c, nil /* path */, visited); err != nil {
+			return errWrapf(err, "this function introduces a cycle")
```

Comment:
The original error message is for a Provide, referencing the function that was
provided. In this case, the error can probably be "found a cycle" or something
similarly generic.